### DISCUSSION TITLE
Artifact update problem for npm project that have private github repo deps

### DISCUSSION THREAD

**Discussion by vtolstov** [renovatebot/renovate#8789] [Request Help]
```
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t https://github.com/unistack-org/cms-typescript-sdk.git
npm ERR! 
npm ERR! remote: Invalid username or password.
npm ERR! fatal: Authentication failed for 'https://github.com/unistack-org/cms-typescript-sdk.git/'
npm ERR! 
npm ERR! exited with error code: 128

npm ERR! A complete log of this run can be found in:
npm ERR!     /tmp/renovate-cache/others/npm/_logs/2021-02-20T23_55_10_348Z-debug.log
```

i'm already have renovate hostRules like
```
 "hostRules": [                                                                                                                                       
    {                                                                                                                                                  
      "hostType": "github",                                                                                                                            
      "encrypted": {                                                                                                                                   
        "token": "IpDd7Zz7s4Kl7msIz4GeWWpoz5Qd+ipEqquCI5RE9iK/kD9E+s/b0GK7IfHUDXyNaMxe/w+cm4XBU/qWY/DMYTjZ6x3hzFwIUOR/pztP2LJgxet2f82qphYeCYBw+        544wyQYvVqzdbVyOEnR7MQyK6jgvMgpfXk+QifcyjJhY4ks4cboKkFM2nSrXX3Huqa9KNyrm7xK+2BRfUicQ1nrf9+vysgZtc+euveW0Ra8oE2zu2qolkf74W/W3Z087NS1FA9kLmyliymmDIG+    pyJyqIVZmAzWwtEZR9qK74XbA0ehH6Ra/JkTlhBRfZaiMi+7yeg4/dzHmA4CQoNMtSAV+Q=="                                                                              
      }                                                                                                                                                
    }                                                                                                                                                  
  ],
```

**Comment by rarkins** [comment]
I will assume you are using the [hosted app](https://github.com/apps/renovate)

Please remove the host rule, as that could cause problems - the app already has its own token.

Now tick the rebase/retry checkbox in the PR and wait for it to complete (you will know when the checkbox is unticked again).

Then view the logs on app.renovatebot.com for the job when the PR was updated (most likely the latest job if you ticked the checkbox to trigger it) and locate the log message "packageFiles with updates". Find the entry for this dependency and paste it in here.

**Reply by vtolstov** [reply]
this lines
```
{
            "depType": "dependencies",
            "depName": "cms-typescript-sdk",
            "currentValue": "git+https://github.com/unistack-org/cms-typescript-sdk.git#master",
            "skipReason": "unversioned-reference",
            "prettyDepType": "dependency",
            "lockedVersion": null,
            "depIndex": 15,
            "updates": []
          },
```

**Reply by rarkins** [reply]
Sorry, I can't help you any further with the incomplete information you've provided.

**Reply by vtolstov** [reply]
i'm already provide log with this dep from app.renovatebot.com

**Reply by vtolstov** [reply]
also i have the same problems with go based sources
```
DEBUG: Datasource 404
{
  "datasource": "go",
  "lookupName": "go.unistack.org/cms-go-sdk",
  "url": "https://api.github.com/repos/unistack-org/cms-go-sdk.git/tags?per_page=100"
}
DEBUG: Failed to look up dependency go.unistack.org/cms-go-sdk (go.unistack.org/cms-go-sdk)(packageFile="go.mod", dependency="go.unistack.org/cms-go-sdk")
```

but for go i have custom host go.unistack.org that returns meta headers to forward sources from github. but as you see deps are not able to fetch.

may be this must be noted - i have free github account with private repos.

**Reply by rarkins** [reply]
Please create a reproduction repo (private) with just *one* dependency that shows the first problem. Provide the full debug log here, after any confidential redaction you need. Do not mix multiple dependencies or problems in the same repository or question here. Describe what you expect to happen.

**Reply by vtolstov** [reply]
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/unistack-org/cms-typescript-sdk.git
npm ERR! 
npm ERR! Host key verification failed.
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR! 
npm ERR! exited with error code: 128

npm ERR! A complete log of this run can be found in:
npm ERR!     /tmp/renovate-cache/others/npm/_logs/2021-02-24T09_46_15_021Z-debug.log

full gist: https://gist.github.com/d0f684a2bc4313c052c15069cb90249f

**Reply by vtolstov** [reply]
I'm experimenting to set dep in package.json via git+https or via github format with owner/repo#tag format. as i saw renovate dont work with private  repos in github. may be this is problem because github not provide GITHUB_TOKEN that allows access to other repos inside org. but i don't understand why hostRules not works

**Reply by rarkins** [reply]
If both repos are inside the same org/account, and you have installed the Renovate app into both repos, and both are private, then the token that's used should have access rights

**Reply by vtolstov** [reply]
both repos inside the same org/account and both of them have configured renovate , ant both are private. But this is not works.

**Reply by rarkins** [reply]
Anything over git/ssh won't work. First step is to get it trying over https. But also I don't know what you're exactly trying to update because the information you've shared is not sufficient

**Reply by vtolstov** [reply]
https://gist.github.com/7e2ce50acd63918461701d7d99c8350f
this full log and repo added to package.json via git+https

**Reply by rarkins** [reply]
There's 120 dependencies in that log - very difficult to debug. Can you reproduce the problem using a single dependency as described in our [minimal reproductions doc](https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md)?
